File: 242-valid-anagram/valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    
  if (s.length !== t.length) return false;
  const seenS = {};
  for (let i = 0; i < s.length; i++) {
    if (seenS[s[i]]) {
      seenS[s[i]]++;
    }
    if (!seenS[s[i]]) {
      seenS[s[i]] = 1;
    }
  }

  for (let i = 0; i < t.length; i++) {
    if (!seenS[t[i]]) return false;
    seenS[t[i]]--;
  }
  for (let i = 0; i < Object.values(seenS).length; i++) {
    if (Object.values(seenS)[i] !== 0) {
      return false;
    }
  }
  return true;
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
    char[] sArr = s.toCharArray();
    char[] tArr = t.toCharArray();
    if (sArr.length != tArr.length)
      return false;

    HashMap<Character, Integer> letters = new HashMap<Character, Integer>();
    for (int i = 0; i < sArr.length; i++) {
      letters.put(sArr[i], letters.getOrDefault(sArr[i], 0) + 1);
    }

    for (int i = 0; i < tArr.length; i++) {
      letters.put(tArr[i], letters.getOrDefault(tArr[i], 0) - 1);
    }
    for (int count : letters.values()) {
      if (count != 0) {
        return false;
      }
    }
    return true; 
    }
}